pandas/core/indexes/datetimelike.py

```diff
@@ -97,6 +97,53 @@ class TimelikeOps(object):
         ValueError if the freq cannot be converted
         """)
 
+    _floor_doc = (
+        """
+        Floor the DatetimeIndex to the specified freq.
+
+        All the available freq can be found in the following link
+        in the offset-aliases section:
+        http://pandas.pydata.org/pandas-docs/stable/timeseries.html
+
+        Parameters
+        ----------
+        freq : str
```

Comment:
Can also be an object: https://pandas.pydata.org/pandas-docs/stable/timeseries.html#dateoffset-objects